engineering/standards/frontend.md

```diff
@@ -326,6 +326,104 @@ TypeScript types (including interfaces) which describe API requests and response
+
+    - ✅ Use props instead.
+
+        ```svelte
+        <script lang="ts">
+          export onValueChange: (value: number) => void;
+
+          onValueChange(42);
+        </script>
+        ```
+
+- Then, in the consuming (parent) component:
+
+    - ❌ Events would be handled as follows:
+
+        ```svelte
+        <script lang="ts">
+          import Child from `./Child.svelte`;
+
+          function handleValueChange({ detail: value }: CustomEvent<number>) {
+                                  // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Yuck!
+            console.log(`Child reports that its value is: ${value}`)
+          }
+        </script>
+
+        <Child on:valueChange={handleValueChange}/>
+        ```
+
+    - ✅ With props:
+
+        ```svelte
+        <script lang="ts">
+          import Child from `./Child.svelte`;
+
+          function handleValueChange(value: number) {
+                                  // ^^^^^^^^^^^^^ Nicer!
+            console.log(`Child reports that its value is: ${value}`)
+          }
+        </script>
+
+        <Child onValueChange={handleValueChange}/>
+        ```
+
+- Rationale
+
+    | | Events | Props |
+    | --  | -- | -- |
+    | Synchronicity | ❌ Events are always synchronous. There's no way to await the `dispatch` call. | ✅ Props give us the ability to define asynchronous callbacks and then await them within the component. We do this within `SpinnerButton` for example. |
+    | Dynamic passing | ❌ Event callbacks [cannot be passed dynamically](https://stackoverflow.com/questions/60934557/how-to-bind-events-dynamically-in-svelte) to dynamically rendered components with `<svelte:component>`. This makes it impossible to do things like perform imperative logic when we're using a `ComponentAndProps` object. | ✅ Props callbacks are easily passed via spreading `{...props}` |
+    | Forwarding | ❌ Event callbacks [cannot be forwarded en masse to child components](https://github.com/sveltejs/svelte/issues/2837). Instead, each event must be manually forwarded. This makes composition more arduous. | ✅ Props callbacks are easily forwarded en masse to child components via `{...$$restProps}` |
+    | Callback parameters | ❌ Events require destructuring the `detail` property when defining callback functions that accept values. This is extra syntax, especially in the type signature of the callback function. | ✅ Props allow us to defined the callback function parameters as we see fit, with no destructuring needed. This is cleaner. |
+    | Required callbacks | ❌ Events are always optional. There is no way to enforce consumers to implement callback functions. This can lead to bugs when someone writing a child component assumes that all parents will implement the callback function. | ✅ Props can be required through the type system. In fact, they are required _by default_. Extra code is necessary to make them optional. The strict defaults make it less bug-prone because error are caught at compile time. |
```

Comment:
This point again confuses a callback prop from an event. If the underlying component needs to have the knowledge of how and when functions in parents need to be called, then it requires a prop. An event is for an entirely different purpose.